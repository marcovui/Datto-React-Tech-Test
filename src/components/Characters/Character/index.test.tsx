import React from 'react';
import { shallow } from 'enzyme';

import { Character } from './index';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('<Character/>', () => {
  test('It renders correctly', () => {
    const props = {
      id: 0,
      name: 'my name',
      birthYear: 'my date of birth',
      gender: 'my gender',
      height: 182,
      notes: [],
    };
    const target = shallow(<Character {...props} />);
    expect(target).toMatchSnapshot();
  });
});
