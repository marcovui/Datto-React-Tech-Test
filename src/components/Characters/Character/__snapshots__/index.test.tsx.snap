// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character/> It renders correctly 1`] = `
<li
  className="character"
>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <div
        className="character__entry"
      >
        <label
          className="character__label"
          htmlFor="name-0"
        >
          Name:
        </label>
        <input
          className="form-control character__input"
          defaultValue="my name"
          id="name-0"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="character__entry"
      >
        <label
          className="character__label"
          htmlFor="birth-year-0"
        >
          Date of birth:
        </label>
        <input
          className="form-control character__input"
          defaultValue="my date of birth"
          id="birth-year-0"
          name="birthYear"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="character__entry"
      >
        <label
          className="character__label"
          htmlFor="gender-0"
        >
          Gender:
        </label>
        <select
          className="form-control character__input"
          defaultValue="female"
          id="gender-0"
          name="gender"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <option
            value="female"
          >
            female
          </option>
          <option
            value="male"
          >
            male
          </option>
          <option
            value="n/a"
          >
            n/a
          </option>
        </select>
      </div>
      <div
        className="character__entry"
      >
        <label
          className="character__label"
          htmlFor="height-0"
        >
          Height:
        </label>
        <input
          className="form-control character__input"
          defaultValue={182}
          id="height-0"
          name="height"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
      </div>
    </fieldset>
    <div
      className="btn__group btn__group--inverse"
    >
      <span
        className="character__updated"
      >
        Character updated successfully!
      </span>
      <button
        className="btn btn--edit"
        type="submit"
      >
        Edit
      </button>
      <button
        className="btn btn--delete"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
    </div>
  </form>
  <Notes
    id={0}
    notes={Array []}
  />
</li>
`;
